### PR TITLE
PM-3925 - Tech Debt - Add missed tests for SSO Email 2FA Tokenable 

### PR DESCRIPTION
**Pull request by JaredSnider-Bitwarden**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Resolve [PM-3925](https://bitwarden.atlassian.net/browse/PM-3925):  add tests that I missed adding as part of fixing [PM-1196](https://bitwarden.atlassian.net/browse/PM-1196) in PR #2874. 

## Code changes

* **src/Core/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenable.cs:** Tweak lifetime definition logic to publicly expose token lifetime to improve testability (token lifetime maintained in one location)
*  **test/Core.Test/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenableTests.cs:** Add tests!

## Before you submit

- Please check for formatting errors (`dotnet format --verify-no-changes`) (required)
- If making database changes - make sure you also update Entity Framework queries and/or migrations
- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team

[PM-3925]: https://bitwarden.atlassian.net/browse/PM-3925?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PM-1196]: https://bitwarden.atlassian.net/browse/PM-1196?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: test/Core.Test/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenableTests.cs [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    [Fact]
+    public void Constructor_CustomExpirationDate_ExpirationMatchesProvidedValue()
+    {
+        var customExpiration = DateTime.UtcNow.AddHours(3);
+        var token = new SsoEmail2faSessionTokenable
+        {
+            ExpirationDate = customExpiration
+        };
+
+        Assert.True(Math.Abs((customExpiration - token.ExpirationDate).TotalMilliseconds) < _timeTolerance.TotalMilliseconds);
```
Comment (justindbaur) [inline test/Core.Test/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenableTests.cs:62]: I believe you can instead do: `(customExpiration - token.ExpirationDate).Duration() < _timeTolerance` to me at least, that feels more readable.

File: test/Core.Test/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenableTests.cs [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+using Bit.Core.Entities;
+using Bit.Core.Tokens;
+using Xunit;
+namespace Bit.Core.Test.Auth.Models.Business.Tokenables;
+
+// Note: these test names follow MethodName_StateUnderTest_ExpectedBehavior pattern.
+public class SsoEmail2faSessionTokenableTests
+{
+    // Allow a small tolerance for possible execution delays or clock precision to avoid flaky tests.
+    private readonly TimeSpan _timeTolerance = TimeSpan.FromMilliseconds(10);
```
Comment (justindbaur) [inline test/Core.Test/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenableTests.cs:12]: `private static readonly`

**Review by justindbaur — CHANGES_REQUESTED** [review]
Good stuff, I always love tests!

File: src/Core/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenable.cs [hunk 3]
```diff
@@ -6,11 +6,13 @@ namespace Bit.Core.Auth.Models.Business.Tokenables;
...
 // This token just provides a verifiable authN mechanism for the API service
 // TwoFactorController.cs SendEmailLogin anonymous endpoint so it cannot be
-// used maliciously. 
+// used maliciously.
 public class SsoEmail2faSessionTokenable : ExpiringTokenable
 {
     // Just over 2 min expiration (client expires session after 2 min)
-    private static readonly TimeSpan _tokenLifetime = TimeSpan.FromMinutes(2.05);
+    private const double _tokenLifetimeInMinutes = 2.05;
+    public static TimeSpan GetTokenLifetime() => TimeSpan.FromMinutes(_tokenLifetimeInMinutes);
```
[the remark this answers is not in the document]
Comment (JaredSnider-Bitwarden) [reply src/Core/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenable.cs]: Fair! Removed const.

File: test/Core.Test/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenableTests.cs (hunk 1 above)
Comment (JaredSnider-Bitwarden) [reply test/Core.Test/Auth/Models/Business/Tokenables/SsoEmail2faSessionTokenableTests.cs]: Ah, that is much cleaner! Thank you!

**Review by justindbaur — APPROVED** [review]

**Pull request merged**